modules/apps/frontend-data-set/frontend-data-set-taglib/src/main/java/com/liferay/frontend/data/set/taglib/internal/jaxrs/application/FDSApplication.java

```diff
@@ -70,6 +71,45 @@
 )
 public class FDSApplication extends Application {
 
+	@DELETE
+	@Path("/fds/{fdsName}/custom-views/{customViewId}")
+	public Response deleteFDSCustomView(
+		@PathParam("fdsName") String fdsName,
+		@PathParam("customViewId") String customViewId,
+		@Context HttpServletRequest httpServletRequest,
+		@Context ThemeDisplay themeDisplay) {
+
+		try {
+			PortalPreferences portalPreferences =
+				PortletPreferencesFactoryUtil.getPortalPreferences(
+					httpServletRequest);
+
+			String fdsSettingsNamespace =
+				ServletContextUtil.getFDSSettingsNamespace(
+					httpServletRequest, fdsName);
+
+			JSONObject customViewsJSONObject = _jsonFactory.createJSONObject(
+				portalPreferences.getValue(
+					fdsSettingsNamespace, "customViews", "{}"));
+
+			customViewsJSONObject.remove(customViewId);
+
+			portalPreferences.setValue(
+				fdsSettingsNamespace, "customViews",
+				customViewsJSONObject.toString());
+
+			return Response.ok(
+			).build();
+		}
+		catch (Exception exception) {
+			_log.error(exception);
+		}
+
+		return Response.status(
+			Response.Status.INTERNAL_SERVER_ERROR
```

Comment:
Maybe in the future, we could check the possible errors and return a proper description and error. I assume that this endpoint is for internal use so we could change the response and the clients will not suffer a breaking change in the response. 
If not, we should check the errors to return the proper information.

Reply:
Yes, endpoint is for internal use, so this is not a blocker for this PR. But, it's a good suggestion! I think we can extend error reporting as a separate task, I added the task to [technical improvements checklist](https://liferay.atlassian.net/wiki/spaces/ENGFRONTENDINFRA/pages/2002681985/FDS+technical+improvements+checklist#FDS-API).